### ISSUE TITLE
Error "list index out of range" when downloading from Quotev

### ISSUE THREAD

**Issue by amyspark** [JimmXinu/FanFicFare#115]
http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/1

FanFicFare version: 2.2.16

Judging from the task log, it would seem that the epilogue chapter is not recognised.

```log
FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(42):FanFicFare v2.2.16
FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(46):Adding jobs for URLs:
FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(48):http://www.quotev.com/story/1746542
FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(84):Logfile for book ID None (I never thought you'd be in my life)
FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(85):url:(http://www.quotev.com/story/1746542) id:(None)
FFF:DEBUG:calibre_plugins.fanficfare_plugin.jobs(151):cookiejar:<LWPCookieJar[]>
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(282):#####################################
pagecache HIT: http://www.quotev.com/story/1746542
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(282):#####################################
pagecache HIT: http://i.quotev.com/img/q/u/12/06/07/bandofoutsiders_ron_harry_potter.jpg
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.story(92):image used unchanged
FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(217):write to C:\Users\Amalia\AppData\Local\Temp\calibre_whcf_8\fanficfare_xtn2bf\I never thought you'd be in my life--ewdyn8.epub
FFF:INFO:calibre_plugins.fanficfare_plugin.fanficfare.writers.base_writer(179):Save directly to file: C:\Users\Amalia\AppData\Local\Temp\calibre_whcf_8\fanficfare_xtn2bf\I never thought you'd be in my life--ewdyn8.epub
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/1
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/2
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/3
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/4
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/5
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/6
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/7
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/8
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/9
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/10
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/11
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/12
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/13
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/14
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/15
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/16
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/17
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(219):code failed:utf8
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(219):code failed:Windows-1252
FFF:INFO:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(221):Could not decode story, tried:['utf8', 'Windows-1252'] Stripping non-ASCII.
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/18
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/19
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/20
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/21
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/22
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/23
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/24
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/25
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/26
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/27
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/28
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/29
FFF:DEBUG:calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter(292):#####################################
pagecache MISS: http://www.quotev.com/story/1746542/I-never-thought-youd-be-in-my-life/30
FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(302):Exception: {u'title': u"I never thought you'd be in my life", u'tags': [u'Scorpius', u'Harry Potter', u'Fan Fiction', u'Romance', u'Malfoy', u'Short Story', u'Love', u'Scorose', u'Short Stories', u'Weasley', u'Rose', u'Completed', u'FanFiction'], u'series': '', u'password': '', u'all_metadata': {'language': '', 'title': u"I never thought you'd be in my life", 'category': u'Fan Fiction, Harry Potter, Love, Malfoy, Romance, Rose, Scorose, Scorpius, Short Stories, Short Story, Weasley', 'storyUrl': u'http://www.quotev.com/story/1746542', 'version': u'FanFicFare v2.2.16', 'extratags': u'FanFiction', 'dateCreated': u'2015-12-27 19:52:40', 'rating': '', 'formatname': '', 'ships': u'', 'genre': u'', 'series': '', 'favorites': u'84', 'storyId': u'1746542', 'pages': '', 'formatext': '', 'numWords': '', 'status': u'Completed', 'reads': '', 'seriesHTML': '', 'authorId': u'', 'siteabbrev': u'quotev.com', 'numChapters': u'30', 'characters': u'', 'langcode': '', 'description': u'<div class="nb_link" id="qdesct">A Scorpius x Rose fanfic</div>', 'warnings': u'', 'site': u'quotev.com', 'datePublished': u'2012-06-07', 'readers': '', 'authorHTML': u'', 'replace_tags_with_spans': u'', 'keep_html_attrs': u'', 'seriesUrl': '', 'authorUrl': u'', 'author': u'', 'lastupdate': u'Last Update Year/Month: 2012/06, Last Update: 2012/06/07', 'output_css': '', 'cover_image': u'specific', 'dateUpdated': u'2012-06-07'}, u'publisher': u'quotev.com', u'begin': None, u'is_adult': False, u'listorder': 0, u'status': u'Error', u'pubdate': datetime.datetime(2012, 6, 7, 20, 39, 39, tzinfo=SafeLocalTimeZone()), u'username': 'NoneGiven', u'icon': u'dialog_error.png', u'url': u'http://www.quotev.com/story/1746542', u'author_sort': [], u'calibre_columns': {}, u'good': False, u'timestamp': datetime.datetime(2015, 12, 27, 19, 52, 49, 56000, tzinfo=SafeLocalTimeZone()), u'outfile': u"C:\\Users\\Amalia\\AppData\\Local\\Temp\\calibre_whcf_8\\fanficfare_xtn2bf\\I never thought you'd be in my life--ewdyn8.epub", u'comment': u'list index out of range', u'updatedate': datetime.datetime(2012, 6, 7, 21, 8, 25, tzinfo=SafeLocalTimeZone()), u'added': False, u'end': None, u'author': [], u'calibre_id': None, u'site': 'quotev.com', u'comments': u'<p>A Scorpius x Rose fanfic</p>'}:list index out of range
Traceback (most recent call last):
  File "calibre_plugins.fanficfare_plugin.jobs", line 219, in do_download_for_worker
  File "calibre_plugins.fanficfare_plugin.fanficfare.writers.base_writer", line 227, in writeStory
  File "calibre_plugins.fanficfare_plugin.fanficfare.writers.writer_epub", line 333, in writeStoryImpl
IndexError: list index out of range

FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(94):
Download Results:
http://www.quotev.com/story/1746542 list index out of range

FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(96):
Successful:


FFF:INFO:calibre_plugins.fanficfare_plugin.jobs(97):
Unsuccessful:
http://www.quotev.com/story/1746542
```

**Comment by JimmXinu** [comment]
Apparently quotev.com now allows anonymous stories.  Change 1553bd2faccce8251d03128fcbd6fc4f495f496f

See [posted test version](http://www.mobileread.com/forums/showthread.php?p=3084025#post3084025) if you'd like to try it before the next release.

**Comment by amyspark** [comment]
It's fixed with 1553bd2. Thanks!
